frontend/src/component/Activity/MemberActivity.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import { translate } from 'react-i18next'
+import { escape as escapeHtml } from 'lodash'
+
+import {
+  Avatar,
+  DistanceDate,
+  TLM_ENTITY_TYPE as TLM_ET,
+  TLM_CORE_EVENT_TYPE as TLM_CET,
+  ROLE_LIST,
+  SUBSCRIPTION_TYPE_LIST
+} from 'tracim_frontend_lib'
+
+require('./MemberActivity.styl')
+
+export class MemberActivity extends React.Component {
+  getSpaceMemberText (coreEventType, i18nOpts) {
+    const { props } = this
+    const role = ROLE_LIST.find(r => r.slug === props.activity.newestMessage.fields.member.role)
+
+    i18nOpts = {
+      ...i18nOpts,
+      role: props.t(role.label)
+    }
+    switch (coreEventType) {
+      case TLM_CET.CREATED:
+        return props.t('{{user}} joined the space', i18nOpts)
+      case TLM_CET.MODIFIED:
+        return props.t("{{user}}'s role has been changed to {{role}}", i18nOpts)
+    }
+  }
+
+  getSpaceSubscriptionText (coreEventType, i18nOpts) {
+    const { props } = this
+    const subscription = SUBSCRIPTION_TYPE_LIST.find(s => s.slug === props.activity.newestMessage.fields.subscription.state)
+    i18nOpts = {
+      ...i18nOpts,
+      state: props.t(subscription.label)
+    }
+    switch (coreEventType) {
+      case TLM_CET.CREATED:
+        return props.t('{{user}} wants to join the space', i18nOpts)
+      case TLM_CET.MODIFIED:
+        return props.t("{{user}}'s request to join the space has been {{state}}", i18nOpts)
+    }
+  }
+
+  getText () {
+    const { props } = this
+    const userPublicName = escapeHtml(props.activity.newestMessage.fields.user.public_name)
+    const [entityType, coreEventType] = props.activity.newestMessage.event_type.split('.')
+    const i18nOpts = {
+      user: `<span title='${userPublicName}' className='member_activity__user'>${userPublicName}</span>`,
+      interpolation: { escapeValue: false }
+    }
+    switch (entityType) {
+      case TLM_ET.SHAREDSPACE_MEMBER:
+        return this.getSpaceMemberText(coreEventType, i18nOpts)
+      case TLM_ET.SHAREDSPACE_SUBSCRIPTION:
+        return this.getSpaceSubscriptionText(coreEventType, i18nOpts)
+    }
+    return <span>Unknown entity type</span>
+  }
+
+  render () {
+    const { props } = this
+    const newestMessage = props.activity.newestMessage
+    return (
+      <div className='member_activity'>
+        <div className='member_activity__left'>
+          <Avatar publicName={newestMessage.fields.user.public_name} width={32} style={{ marginRight: '5px' }} />
+          <div dangerouslySetInnerHTML={{ __html: this.getText() }} />
+        </div>
+        <div className='member_activity__right'>
+          <DistanceDate absoluteDate={newestMessage.created} lang={props.user.lang} /><br />
+          {props.t('by')} <span className='member_activity__user'>{newestMessage.fields.author.public_name}</span>
```

Comment:
We have that same piece of code in different places. Shouldn't we make a "Last Modification" component or something like that?

Reply:
I was thinking about ContentActivityHeader not NotiicationWall, but you are right that they are a little different. My idea was to props the type of modification (not displayed if null, like for member), the author's name, the language and the date and make this display in two lines with the author in bold. In the future this same style can also be used in the search page and in the "my favorites" page.

Ps: Did you do the test with a big author's name? if we are going to do the overflow with ellipsis then it is necessary to add a tooltip here (in my opinion you can add it even if the overflo is managed in another way)

Reply:
OK, I created the `TimedEvent` component and checked its behavior with a big author name.